src/GraphType.jl

```diff
@@ -242,10 +253,13 @@ mutable struct Graph
         extra_uuids = union(keys(reqs), keys(fixed), map(fx->keys(fx.requires), values(fixed))...)
         extra_uuids ⊆ keys(versions) || error("unknown UUID found in reqs/fixed") # TODO?
 
-        data = GraphData(versions, deps, compat, uuid_to_name, verbose = verbose)
+        data = GraphData(versions, deps, compat, uuid_to_name, verbose = verbose)::GraphData
         pkgs, np, spp, pdict, pvers, vdict, rlog = data.pkgs, data.np, data.spp, data.pdict, data.pvers, data.vdict, data.rlog
 
-        extended_deps = [[Dict{Int,BitVector}() for v0 = 1:(spp[p0]-1)] for p0 = 1:np]
+        local extended_deps
+        let spp = spp
+            extended_deps = [Vector{Dict{Int,BitVector}}(spp[p0]-1) for p0 = 1:np]::Vector{Vector{Dict{Int, BitVector}}}
+        end
```

Comment:
What's the reason for using a `let` block here?

(also, extra space between `Int` and `BitVector`)

Reply:
See https://github.com/JuliaLang/Pkg3.jl/pull/115#discussion_r166598165

Reply:
The type assertion is because inference couldn't figure out the type of `extended_deps` so it poisons other variables derived from this one.

Reply:
Sorry, I see now that it's due to https://github.com/JuliaLang/julia/issues/15276. Maybe add a comment so that when that is eventually fixed we can revert?